ros_buildfarm/config/__init__.py

```diff
@@ -67,40 +67,45 @@ def get_distribution_file(index, rosdistro_name, build_file):
 def get_ci_build_files(index, dist_name):
     data = _get_build_file_data(index, dist_name, 'ci_builds')
     build_files = {}
-    for k, v in data.items():
+    for (k, url), v in data.items():
         build_files[k] = CIBuildFile(k, v)
+        build_files[k].url = url
     return build_files
 
 
 def get_release_build_files(index, dist_name):
     data = _get_build_file_data(index, dist_name, 'release_builds')
     build_files = {}
-    for k, v in data.items():
+    for (k, url), v in data.items():
         build_files[k] = ReleaseBuildFile(k, v)
+        build_files[k].url = url
```

Comment:
It may be clearer for me to modify the dict's value to be a tuple rather than making the key a tuple. Let me give that a shot.

Reply:
Ah I see where I confused myself. Due to the diff folding on GitHub I misread the change in `_load_build_file_data` as just another iteration of the return from that same helper rather than modifying the value of `data` for each of the above uses of it. That's what was breaking me, I thought that the key was always a tuple value but being newly destructured not changed to a tuple and destructured.

That being said, I _do_ think that a tuple-valued dict is a lot more readable than a tuple-keyed dict so while I'm sorry for causing you grief I think we've landed in a better place.